npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/yarn_lockfile_updater.rb

```diff
@@ -326,6 +326,17 @@ def write_lockfiles
           end
         end
 
+        def lock_deps_with_latest_reqs(content)
+          package_files.each do |file|
+            NpmAndYarn::FileParser::DEPENDENCY_TYPES.each do |t|
```

Comment:
Looks like this will parse the same package.json several times (for each dependency type, e.g. `depedencies`, `optionalDependencies`).

We should move the json parse before this loop:

```ruby
package_files.each do |file|
  json = JSON.parse(file.content)
  NpmAndYarn::FileParser::DEPENDENCY_TYPES.each do |t|
  #...
end

Reply:
👍